### ISSUE TITLE
Grafana failed to load its application files

### ISSUE THREAD

**Issue by atropos112** [grafana/grafana#65759]
**What happened**:
While running the newes version of Vivaldi (Vivaldi 5.7.2921.65 stable) and Grafana (9.4.3) I started getting 

```
If you're seeing this Grafana has failed to load its application files


1. This could be caused by your reverse proxy settings.

2. If you host grafana under subpath make sure your grafana.ini root_url setting includes subpath. If not using a reverse proxy make sure to set serve_from_sub_path to true.

3. If you have a local dev build make sure you build frontend using: yarn start, yarn start:hot, or yarn build

4. Sometimes restarting grafana-server can help

5. Check if you are using a non-supported browser. For more information, refer to the list of [supported browsers](https://grafana.com/docs/grafana/latest/installation/requirements/#supported-web-browsers).
```

I tried on a separate computer, using browser in the "private mode" without extensions and seem to be getting the same output **almost** always. By almost I mean that I have periods when I don't get it but those don't last long (usually couple refreshes and I am back to the message above).

**What you expected to happen**:
I expected it to work as before and let me use Grafana.

**How to reproduce it (as minimally and precisely as possible)**:
Install the newest Vivaldi and Grafana version that matches above (I think that's newest also?) and try to access the login screen of your grafana instance via https (not sure if you need https to trigger this)

**Environment**:
- Grafana version: 9.4.3
- OS Grafana is installed on: Ubuntu 22.10
- User OS & Browser: Vivaldi 5.7.2921.65 stable
- Grafana plugins: No self added plugins 

EDIT:
I Checked on Grafana 9.4.2 and same result is happening. It is possible its an update from Vivaldi's side that caused this.

**Comment by usmangt** [comment]
Hi @atropos112,

Thanks for opening this issue. We have the following [list](https://grafana.com/docs/grafana/latest/setup-grafana/installation/#supported-web-browsers) of supported browsers for Grafana and this one is not on the list. Therefore closing this issue

**Comment by atropos112** [comment]
I saw the recommended browsers section. My concern here is about the fact that a chromium based browser previously worked (for months) and now it no longer does, is this intentional or simply "not supported hence do not care if it was recently broken"

**Comment by ict-one-nl** [comment]
I have been running Grafana behind an nginx reverse proxy with the same setup for years. Upgraded to 9.4.7 and now I'm getting the same error most of the time(?), but when I bypass the nginx proxy it does work. In my case it seems less related to the browser; using the web console of the browser, every request gives a 200 OK and still it complains.

**Comment by mbwmbw1337** [comment]
I am seeing the same issue across numerous browsers impacting hundreds of our users.

**Comment by ict-one-nl** [comment]
Solution to my problem was to disable the javascript minify from Cloudflare. Difficult problem to track and to solve, even after disabling the option because of the agressive browser cache.

**Comment by Merrick28** [comment]
@ict-one-nl you made my day ! It worked, thank you.

**Comment by i-gri** [comment]
@ict-one-nl, yep the Cloudflare minify was the culprit for me too. Thank you!

**Comment by mbwmbw1337** [comment]
Same on my side! Thank you!

**Comment by Asymptotezz** [comment]
@ict-one-nl It works! I spent the whole afternoon looking for errors in nginx configs. Thank you so much!

**Comment by ClimberJ** [comment]
I have the same issue except I am not using Cloudflare.
Here are my  configs:
```
        location /grafana {
                limit_req   zone=one  burst=10 nodelay;
                proxy_pass http://192.168.178.114:3000;
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
        }
```
```
;root_url = %(protocol)s://%(domain)s:80/grafana
;serve_from_sub_path = true
```

**Comment by ClimberJ** [comment]
I have thought about that, but the problem is that I am hosting from my home, so I can only have my main domain going to my server.

**Comment by ict-one-nl** [comment]
You are aware of SNI? You can host multiple TLS hosts on one IP. I even put cloudflare in front of mine, but that's optional.

**Comment by ClimberJ** [comment]
I was not. I am not sure how SNI would work with my current network setup, though, as I am currently routing my traffic like this:
Internet > router w/firewall > WAF > nginx reverse proxy > application.

**Comment by ict-one-nl** [comment]
I'm doing the same, no problem. Cloudflare - router - nginx reverse proxy - applicaties (within Docker network).

**Comment by ClimberJ** [comment]
Turns out the ; at the start of the line in the grafana config makes grafana ignore the line. With this config it worked:
```
root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana
serve_from_sub_path = true
```

**Comment by Pyker** [comment]
> Solution to my problem was to disable the javascript minify from Cloudflare. Difficult problem to track and to solve, even after disabling the option because of the agressive browser cache.

I feel like this needs to be documented somewhere in the Grafana docs. Drove me nuts for 2 days trying to figure out what was going on until I found this one comment hidden away in this one issue.

**Comment by thapakazi** [comment]
after going via a day in this rabbit hole, I also confirmed its cloudflare cache 😭 

> Solution to my problem was to disable the javascript minify from Cloudflare. Difficult problem to track and to solve, even after disabling the option because of the agressive browser cache.

I had to purge my cache, I still wonder why this occurred and how cloudflare cache made it worse 🤔

**Comment by daniel-koudouna** [comment]
> Solution to my problem was to disable the javascript minify from Cloudflare. Difficult problem to track and to solve, even after disabling the option because of the agressive browser cache.

I can confirm this worked for me as well with the cloudflare -> nginx proxy setup.
Thank you!

**Comment by JinjieZhao** [comment]
I encountered this problem, which arises from the Nginx proxy losing messages.
chrome console: GET /grafana/public/build/grafana.light.1d77523c18081c26d59e.css net::ERR_CONTENT_LENGTH_MISMATCH 200 (OK)
nginx error: /nginx/proxy_temp/6/09/0000000096" failed (13: Permission denied) while reading upstream

how to do:
    proxy_buffers 8 4M;
    proxy_busy_buffers_size 4M;

**Comment by Scotee007** [comment]
> 我遇到了这个问题，这是由于Nginx代理丢失消息引起的。 chrome 控制台：获取 /grafana/public/build/grafana.light.1d77523c18081c26d59e.css net：：ERR_CONTENT_LENGTH_MISMATCH 200（正常） nginx 错误：/nginx/proxy_temp/6/09/0000000096“在上游读取时失败（13：权限被拒绝）
> 
> 怎么做：proxy_buffers 8 4M; proxy_busy_buffers_size 4M;

yes, bro. I also find this work. Auto Minify JavaScript actually affects other webservice unless you modify specially your self. So I think your solution is better!!!!
That is also found in a CSDN like this:
https://blog.csdn.net/privateobject/article/details/126368897

**Comment by farzadpanahi** [comment]
We have encountered the same error trying to access Grafana from latest Edge browser running on a Windows Server. There are no proxies set for Windows.  No sign of `X-Forwarded-For` in headers. There is no Cloudflare involved. The whole network is offline (not connected to the Internet). The requests are all 200 or 302. There are console log errors due to syntax errors in .js files. 

I know this issue is closed but for this specific error, are there any causes other than unsupported browsers and proxies?

Basically if we are using the latest up-to-date supported browser and there is no proxy in between the browser and Grafana, what else can cause this error?

**Comment by sachasmart-weavik** [comment]
Running into this same issue - cleared Cloudflare cache and set the appropriate optimization config. No luck.

**Comment by tleerai** [comment]
> after going via a day in this rabbit hole, I also confirmed its cloudflare cache 😭
> 
> > Solution to my problem was to disable the javascript minify from Cloudflare. Difficult problem to track and to solve, even after disabling the option because of the agressive browser cache.
> 
> I had to purge my cache, I still wonder why this occurred and how cloudflare cache made it worse 🤔

Specifically, for me it worked to just purge the hostname: `monitoring.example.com` then in Chrome F12 + right click Refresh button and select empty cache and hard reload.

**Comment by siinghd** [comment]
Any update on this? The above solutions didn't worked for me. 
I'm using nginx and cloudflare.

**Comment by siinghd** [comment]
Update: the problem is with cloudflare or nginx... bypassing both works fine

**Comment by sachasmart-weavik** [comment]
> Update: the problem is with cloudflare or nginx... bypassing both works fine

I didn't have time for this and ended up just serving at as IP in a VPN - to me this is a huge issue and Grafana's docs about hosting behind a reverse proxy (in my case Traefik) was lacking to say the least.

**Comment by CptHolzschnauz** [comment]
Same problem, i'm trying to set up a curl proxy in a php to mitigate the auth process from the iframe to the server with php. 
(Background is i use avout 20 iframes with the panels in a PWA. The Grafana login page appears on every iframe. if atuthentificated in one iframe, i have to restart the whole PWA to be authetifiacated on all iframes, thats why i try this.)

The panel URL is part of the GET request:
```
<?php
session_start();
if (!isset($_SESSION['userid'])) {
    die('<a href="login.php">PLEASE LOG IN</a>  <script type="text/javascript">
    window.location.href = "login.php";
    </script>');
}
?>
<?php
// Grafana credentials
$grafanaUsername = 'proxy';
$grafanaPassword = 'password';

// Validate user credentials (you might implement your own validation logic)
/*
if (!isset($_SERVER['PHP_AUTH_USER']) || !isset($_SERVER['PHP_AUTH_PW']) ||
    $_SERVER['PHP_AUTH_USER'] !== $grafanaUsername || $_SERVER['PHP_AUTH_PW'] !== $grafanaPassword) {
    header('WWW-Authenticate: Basic realm="Restricted Area"');
    header('HTTP/1.0 401 Unauthorized');
    echo 'Unauthorized';
    exit;
}
*/
// Get the Grafana panel URL from the client's GET request
// Get the Grafana panel URL from the client's GET request
$grafanaPanelUrl = isset($_GET['grafanaPanelUrl']) ? $_GET['grafanaPanelUrl'] : '';

if (empty($grafanaPanelUrl)) {
    echo 'Grafana panel URL is missing.';
    exit;
}

// Forward the request to Grafana
$ch = curl_init($grafanaPanelUrl);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true); // Follow redirects

// Forward HTTP headers from the original request to Grafana
foreach (getallheaders() as $key => $value) {
    curl_setopt($ch, CURLOPT_HTTPHEADER, ["$key: $value"]);
}

// Execute the request
$response = curl_exec($ch);

// Check for errors
if (curl_errno($ch)) {
    echo 'Error: ' . curl_error($ch);
    exit;
}

// Forward Grafana's response to the client
echo $response;

// Close cURL session
curl_close($ch);
?>


```
But i get the site with
```

If you're seeing this Grafana has failed to load its application files

1. This could be caused by your reverse proxy settings.

2. If you host grafana under subpath make sure your grafana.ini root_url setting includes subpath. If not using a reverse proxy make sure to set serve_from_sub_path to true.

3. If you have a local dev build make sure you build frontend using: yarn start, or yarn build

4. Sometimes restarting grafana-server can help

5. Check if you are using a non-supported browser. For more information, refer to the list of[ supported browsers](https://grafana.com/docs/grafana/latest/installation/requirements/#supported-web-browsers).
```

So, what i see is that the requests for the Grafana application files are requested from the domain where the iframe is hosted eg from
iframehost.domain.com while the Grafana is under grafana.domain.com

With iframes with direct request to grafana.domain.com it works.
The proxy php is under iframehost.domain.com.
No subpath. 
Question: Is the proxy attempt impossible or it's just my setup?

**Comment by arashkhavari** [comment]
I was using something other than Cloudflare and I had a problem
My reverse proxy was nginx and I had this problem
I solved the problem by putting this option in my location blocks:
`proxy_buffering off;`

**Comment by Wired4ncer** [comment]
Same issue here, but only in Firefox browser. Really can't figure it out. None of the above works. No cloudflare, ubuntu 22 vps with nginx. Chromium and Brave works fine, Firefox throws the error after login.

**Comment by tomwiggers** [comment]
@Wired4ncer Also experiencing this issue since today but I cannot figure out what it is either. I've tried downgrading Grafana from 10.4.2 to 10.3.3 (previous version I was running) but that did not fix it. I've tried clearing cookies, cache, different version of Authelia, messing with Authelia config (I use that for SSO). Interestingly, in Firefox it does not work, but on Vivaldi it actually does (installed Vivalid after reading OP).

**Comment by MrPostie** [comment]
Got the same issue today. Don't know about yesterday, didn't login and also in Firefox (FF). Private window in FF has no issues, tried other browsers and could login. Had 10.4.1 installed, updated to 10.4.2 but that did not solve the issue.
I'm running Grafana behind a Traefik proxy, no sub path. Everything worked fine the last time I logged in with FF.

**Comment by Wobwobwob** [comment]
Same issue, no reverse proxy, no Cloudflare. Chrome works, Firefox almost always fails. Sometimes I get the login page but upon reload or sign-in I get the error again. I'm using the grafana/grafana:latest docker image. Feels like this should be reopened.

**Comment by andreapx** [comment]
I've had this problem this morning and I haven't updated Firefox or Grafana.
I've disable the extension CanvasBlocker and it started working again, but I've no idea why it should impact Grafana.

**Comment by kot0dama** [comment]
Grafana suddenly stopped working that same way for me a few days ago, I tried deactivating some extensions just on Grafana URL, and also Firefox `resistfingerprinting` option  but to no avail.

But then as it worked on a clean firefox profile, I tried disabling all extensions altogether, and re-enabled them one by one, each time reloading Grafana with CTRL+MAJ+R.

For me it was CanvasBlocker indeed, which even when I disable CanvasBlocker through the extension helper icon still has Grafana fail to load.

This is actually https://github.com/kkapsner/CanvasBlocker/issues/707

**Comment by tomwiggers** [comment]
@kot0dama seems you are correct. Whitelisting all APIs does not help, I also need to disable the plugin. Seems like it is https://github.com/kkapsner/CanvasBlocker/issues/707 indeed.

**Comment by spidey** [comment]
For me it was CanvasBlock too. Hard to debug issue, additionally I've changed my domain so I started the investigation from the reverse proxy configuration. Glad it's solved.

**Comment by AxeOfMen** [comment]
It was a Firefox profile issue for me. Running `firefox --ProfileManager` and creating a new profile, I was able to view Grafana as expected. This demonstrates the problem is with my Firefox profile. I reverted to my original profile to rescue it and did the following:

1. Navigate to `about:support` in Firefox and click the "Open Folder" button to open the folder containing your Firefox profile.
2. Remove the following:

- minidumps/
- storage/
- times.json
- xulstore.json

3. Restart Firefox to ensure it works.

After this, I restored the files/folders one by one to figure out which was to blame, but after restoring each of them, Grafana continued to work. So if you're worried any of these contain valuable information to your profile, simply move them to another place, start Firefox, navigate to Grafana, close Firefox and restore the files to your profile.

**Comment by AashishDhakal** [comment]
If you are on CloudFlare and Nginx adding `proxy_buffering off;` to Nginx location blocks will solve the issue.

**Comment by L1nker2** [comment]
I use chromium with the "headless" parameter in docker on the server, I encountered this message when running tests using the library "playwright-python" Can anyone tell me if anyone has encountered this when running tests?

**Comment by mikenye** [comment]
We encountered this same issue in our environment and were able to resolve it.

In our case, the root cause was outdated browsers — specifically Firefox 79 and Edge 88 — which do not support private class fields in JavaScript (e.g. #privateField = ...). These are now used in Grafana's frontend code, likely introduced in recent versions that adopt ECMAScript 2022+ syntax.

Although all assets were being served correctly (no 404s or MIME issues), older browsers failed to parse the JS due to this syntax, resulting in the “Grafana has failed to load its application files” error. Chrome worked fine because it supports this syntax.

Solution: Updating to a more modern browser (e.g. Firefox 100+, Edge 90+, or Chrome 90+) resolved the issue. No changes to Grafana or NGINX were required.

Hope this helps others running into the same problem!

**Comment by jjavieralv** [comment]
Also **checks your disk space**
df -h

Could be that your system has not enough space on the disk and that causes that nginx cant buffer the js that grafana is trying to retrieve. Just remove some files to get more space or increase your main volume. You can check it if you go to your browser, open dev-tools and check if you got a partial retrive of the json. (it is indicated on the error code or if you call that file using curl it will pop a message saying that there are more bytes that can not be retrieved)

**Comment by qcqcainiao** [comment]
`If you're seeing this Grafana has failed to load its application files`

I have encountered this peculiar problem.
My scenario was to deploy Grafana offline and access http://ip:3000 directly without any proxy. 

The Google version is 93.0.4577.82 (an older version)
Grafana is 12.2.1 (the latest version) 

My solution is: Upgrade the Google version to solve it. 

Summary:
Older versions of browsers cannot access the higher version of Grafana.

Solution:
1. Upgrade the version of Google Chrome
2. Lower the version of the Grafana server, you can choose 7.x

**Comment by shenyinabcd** [comment]
I coping with this problems already several days. Until today I found the Edge Browser version is 89.xx. WTF

**Comment by shenyinabcd** [comment]
Update browser version did solve my problem. 

4 working days wasted...
